services/src/unit-test/java/uk/gov/moj/sdt/services/SubmitQueryServiceTest.java

```diff
@@ -423,54 +366,59 @@ public void testSubmitQueryRequestThrottling() {
-        receiveTimeOutParam.setName("TARGET_APP_RESP_TIMEOUT");
+        receiveTimeOutParam.setName(TARGET_APP_RESP_TIMEOUT);
         receiveTimeOutParam.setValue("12000");
-        EasyMock.expect(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "TARGET_APP_RESP_TIMEOUT"))
-                .andReturn(receiveTimeOutParam);
+        when(mockGlobalParamCache.getValue(IGlobalParameter.class, TARGET_APP_RESP_TIMEOUT))
+                .thenReturn(receiveTimeOutParam);
 
         final IGlobalParameter maxQueryReq = new GlobalParameter();
         maxQueryReq.setName("MCOL_MAX_CONCURRENT_QUERY_REQ");
         maxQueryReq.setValue("5");
-        EasyMock.expect(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "MCOL_MAX_CONCURRENT_QUERY_REQ"))
-                .andReturn(maxQueryReq);
-
-        this.mockConsumerGateway.submitQuery(submitQueryRequest, 1000, 12000);
-        EasyMock.expectLastCall().andAnswer(new IAnswer<Object>() {
-            @Override
-            public Object answer() throws Throwable {
-                ((SubmitQueryRequest) EasyMock.getCurrentArguments()[0]).setStatus(ISubmitQueryRequest.Status.OK
-                        .getStatus());
-                // required to be null for a void method
-                return null;
-            }
-        });
+        when(mockGlobalParamCache.getValue(IGlobalParameter.class, "MCOL_MAX_CONCURRENT_QUERY_REQ"))
+                .thenReturn(maxQueryReq);
 
-        EasyMock.replay(mockConsumerGateway);
-        EasyMock.replay(mockGlobalParamCache);
-        EasyMock.replay(mockErrorMsgCacheable);
-        EasyMock.replay(mockBulkCustomerDao);
+        doAnswer(invocation -> {
+            SubmitQueryRequest request = (SubmitQueryRequest) invocation.getArguments()[0];
+            request.setStatus(ISubmitQueryRequest.Status.OK.getStatus());
+            return null;
+        }).when(mockConsumerGateway).submitQuery(submitQueryRequest, 1000, 12000);
+
+        // Set logging level to debug so that afterCompletion logging is captured
+        logger.setLevel(DEBUG);
+
+        // Create appender and add to logger
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
 
         SdtContext.getContext().setRawInXml("criteria");
-        this.submitQueryService.submitQuery(submitQueryRequest);
+        submitQueryService.submitQuery(submitQueryRequest);
+
+        List<ILoggingEvent> logList = listAppender.list;
+
+        assertTrue(verifyLog(logList,"Increment concurrent requests. Current requests in progress"));
 
-        EasyMock.verify(mockConsumerGateway);
-        EasyMock.verify(mockGlobalParamCache);
-        EasyMock.verify(mockErrorMsgCacheable);
-        EasyMock.verify(mockBulkCustomerDao);
+        logger.detachAndStopAllAppenders();
+
+        verify(mockConsumerGateway).submitQuery(submitQueryRequest, 1000, 12000);
```

Comment:
Add code to verify behaviour of other mocked objects.

Reply:
done